### ISSUE TITLE
trial(1) man page says "I don't know why this is in trial." for --disablegc

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4511]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| spiv reported|
|-|-|
|Trac ID|trac#4511|
|Type|defect|
|Created|2010-06-20 12:25:51Z|
|Branch|https://github.com/twisted/twisted/tree/trial-disablegc-man-4511|

```
$ grep -A1 disablegc doc/core/man/trial.1 
\fB--disablegc\fR
Disable the garbage collector.  I don't know why this is in trial.
```

I think I know why this option is in Trial.

 1. It will make some test suites measurably faster (as a test suite runs and accumulates results (and tracebacks for failures), the number of objects tracked by the VM grows, and so the time it takes the VM to run its periodic cyclic GC passes slows.  Disabling the cycle collector avoids this cost (at the cost of possibly greatly increasing memory consumption).  Contrast `--force-gc`.  A better solution is to keep the memory/object consumption of your test suite low in the first place, though.
 1. It can may tests slightly more deterministic, which theoretically can be a useful debugging aid, although I'd probably only try it out of desperation.

Now all it needs is someone to distil that knowledge into something suitable for the man page... ideally without making that option sound particularly important or attractive, as it really is just a convenience for niche situations.

Attachments:

* [document-disablegc-4511.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/87e/87ee54b1820f961fe68a462f7352b9215a965a43/d00e0e293e31c465e1ce20f5cfb4ceefe9007792.patch) (2120 bytes) - added by Screwtape on 2010-06-27 06:01:53Z - Document --disablegc

<details><summary>Searchable metadata</summary>

```
trac-id__4511 4511
type__defect defect
reporter__spiv spiv
priority__normal normal
milestone__ 
branch__branches_trial_disablegc_man_4511 branches/trial-disablegc-man-4511
branch_author__Screwtape Screwtape
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__documentation documentation
time__1277036751000000 1277036751000000
changetime__1277816283000000 1277816283000000
version__None None
owner__ 
cc__Screwtape
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Screwtape's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|Screwtape removed owner|
|-|-|

The attached patch summarizes spiv's description in the `--disablegc` documentation (which should probably be `--disable-gc` to match `--force-gc`, but that's another bug entirely).

Also, `--help-reporters` and `--help-reactors` mention how to use the information they provide, and `--help-reactors`' wording is consistent with `--help` and `--help-reporters`.

Finally, every reference to `trial` uses a lower-case T (not how I'd choose to do it myself, but the overwhelming style of the existing text).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [29415]) Branching to 'trial-disablegc-man-4511'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [29416]) Apply document-disablegc-4511.patch

refs [#4511](https://github.com/twisted/twisted/issues/4511)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [29417]) Two very minor rewordings

refs [#4511](https://github.com/twisted/twisted/issues/4511)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [29418]) wrap long line

refs [#4511](https://github.com/twisted/twisted/issues/4511)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

I made a few trivial edits, linked above.  Otherwise this looks good.  Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [29420]) Merge trial-disablegc-man-4511

Author: Screwtape
Reviewer: exarkun
Fixes: [#4511](https://github.com/twisted/twisted/issues/4511)

Expand the man page documentation for the trial option `--disablegc`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
